atsign-foundation/at_libraries
# feat: Secure atKeys with pass-phrase

**- What I did**
Support the password protected of atKeys file.

**- How I did it**

1. Add pass phrase to the AtOnboardingPreference to capture the pass phrase and propagate it the at_auth and at_chops package to decrypt the atKeys.

2. In auth_cli_args, add parser options to capture "pass phrase" and "hashing algo type"

3. In at_onboarding_service_impl.dart, in "_generateAtKeysFile", check if the passPhrase is supplied. If supplied, encrypt the keys with the passphrase.

4. Removed the at_cli_commons dependency from at_onboarding_cli to prevent from cyclic dependency. 

**- How to verify it**
1. Added a test to verify the encryption and decryption of atKeys using a passphrase with the argon2id algorithm

2. A functional test to verify password protected of atKeys file

**- Description for the changelog**
- feat: Support password protected of atKeys
